### PR TITLE
Small improvements to the HttpStatusPush Reporter

### PR DESCRIPTION
**Pull request by benallard**
* Add an auth parameter
 * Deprecate the user/password one
 * Add a format_fn parameter

### CODE REVIEW

File: master/buildbot/reporters/http.py [hunk 1]
```diff
@@ -97,17 +97,28 @@ def send(self, build):
...
         HttpStatusPushBase.checkConfig(self, **kwargs)
 
-    def reconfigService(self, serverUrl, user, password, **kwargs):
+    def reconfigService(self, serverUrl, user=None, password=None, auth=None, format_fn=None, **kwargs):
         HttpStatusPushBase.reconfigService(self, **kwargs)
         self.serverUrl = serverUrl
-        self.auth = (user, password)
+        self.auth = auth
+        if user is not None:
+            self.auth = (user, password)
```
Comment (sa2ajj) [inline master/buildbot/reporters/http.py]: a slightly cleaner version would be:
```
if user is not None:
    ...
else:
    self.auth = auth
```

File: master/buildbot/reporters/http.py [hunk 2]
```diff
@@ -97,17 +97,28 @@ def send(self, build):
...
+        self.format_fn = format_fn
+        if format_fn is None:
```
Comment (sa2ajj) [inline master/buildbot/reporters/http.py]: same here.

File: master/buildbot/reporters/http.py (hunk 1 above)
Comment (benallard) [inline master/buildbot/reporters/http.py]: One can argue about the "cleaness" of both. The version I wrote made it clear that ``user is not None`` is a special case, and the rest the normal case. In your version, ``user is not None`` is the very first check, and the rest is a fallback, which convey a slightly different meaning. Additionally, from python zen: _Flat is better than nested._.

I'll fix it anyway.

**Comment by benallard** [comment]
... and rebased.

**Comment by tardyp** [comment]
:+1:

**Comment by tardyp** [comment]
I'll let you @sa2ajj merge if you are happy with the changes

**Pull request merged**